File: LeetCode/Medium/0015-3sum/0015-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        // 중복을 피하기 위해 Set 사용
        Set<List<Integer>> results = new HashSet<>();
        Arrays.sort(nums);
        
        for (int i = 0; i < nums.length - 2; i++) {
            int left = i + 1, right = nums.length - 1;
            
            while (left < right) {
                int sum = nums[i] + nums[left] + nums[right];
                
                if (sum < 0) {
                    left++;
                } else if (sum > 0) {
                    right--;
                } else {
                    results.add(Arrays.asList(nums[i], nums[left], nums[right]));
                    left++;
                    right--;
                }
            }
        }
        
        // Set을 리스트로 변환하여 반환
        return new ArrayList<>(results);
    }
}

File: LeetCode/Medium/0015-3sum/0015-3sum.kt
class Solution {
    fun threeSum(nums: IntArray): List<List<Int>> {
        nums.sort()
        val set = mutableSetOf<List<Int>>()

        for (i in 0 until nums.size - 2) {
            var j = i + 1
            var k = nums.size - 1

            while (j < k) {
                val sum = nums[i] + nums[j] + nums[k]

                when {
                    sum < 0 -> j++
                    sum > 0 -> k--
                    else -> {
                        set.add(listOf(nums[i], nums[j], nums[k]))
                        j++
                        k--
                    }
                }
            }
        }

        return set.toList()
    }
}